HW2/P3/mandelbrot.pyx

```diff
@@ -4,50 +4,143 @@ cimport cython
-       #
-       # And then, for example, to multiply them
-       #     v3 = AVX.mul(v1, v2)
-       #
-       # You may find the numpy.real() and numpy.imag() fuctions helpful.
+        int i, j, iter, k
+        np.complex64_t c, z
+        float* tmp_counts
+        AVX.float8 coor_real, coor_imag, z_real, z_imag, new_z_imag, tmp, new_z_real
+        AVX.float8 iterat, mask, add_it, test_z, stop, checker
+        # To declare AVX.float8 variables, use:
+        # cdef:
+        #     AVX.float8 v1, v2, v3
+        #
+        # And then, for example, to multiply them
+        #     v3 = AVX.mul(v1, v2)
+        #
+        # You may find the numpy.real() and numpy.imag() fuctions helpful.
 
     assert in_coords.shape[1] % 8 == 0, "Input array must have 8N columns"
     assert in_coords.shape[0] == out_counts.shape[0], "Input and output arrays must be the same size"
     assert in_coords.shape[1] == out_counts.shape[1],  "Input and output arrays must be the same size"
 
-    with nogil:
-        for i in range(in_coords.shape[0]):
-            for j in range(in_coords.shape[1]):
-                c = in_coords[i, j]
-                z = 0
+    with nogil, parallel(num_threads = 1):
+        tmp_counts =  <float *> malloc(sizeof(float) * 8)
+        for i in prange(in_coords.shape[0], schedule='static', chunksize=1):
+            for j in range(in_coords.shape[1]/8):
+
+                # Note that the order of the arguments here is opposite the direction when
+                # we retrieve them into memory.
+                coor_real = AVX.make_float8(in_coords[i, j*8+7].real,
+                                         in_coords[i, j*8+6].real,
+                                         in_coords[i, j*8+5].real,
+                                         in_coords[i, j*8+4].real,
+                                         in_coords[i, j*8+3].real,
+                                         in_coords[i, j*8+2].real,
+                                         in_coords[i, j*8+1].real,
+                                         in_coords[i, j*8].real)
+
+
+
+                coor_imag = AVX.make_float8(in_coords[i, j*8+7].imag,
+                                         in_coords[i, j*8+6].imag,
+                                         in_coords[i, j*8+5].imag,
+                                         in_coords[i, j*8+4].imag,
+                                         in_coords[i, j*8+3].imag,
+                                         in_coords[i, j*8+2].imag,
+                                         in_coords[i, j*8+1].imag,
+                                         in_coords[i, j*8].imag)
+
+
+
+                z_real = AVX.float_to_float8(0.0)
+
+                z_imag = AVX.float_to_float8(0.0)
+
+                iterat = AVX.float_to_float8(0.0)
+
+                #Stop controls whether the iteration has been broken
+                stop = AVX.float_to_float8(1.0)
+
+
+                #tmp = AVX.float_to_float8(0.0)
```

Comment:
It's good practice to remove unused code.  It helps make the code more readable.